### DISCUSSION TITLE
SFTP on non-default port

### DISCUSSION THREAD

**Discussion by Sythelux** [needle-tools/needle-engine-support#190] [Export: Unity]
I want to deploy my needle project from unity to our sftp server. It is running on a non-default port (as is suggested, when you make SSH on a server on the internet.)

The thing is I can't find a setting to add a port too, when I setup the scriptable object "FTP Server" in Unity. I also can't add the port to the server name.

so what I expect is: either have an extra port field oder the ability to write Servername as: 'example.com:3000' and the deploy to FTP will pick it up.

what is happening is:
I can't put in the port somewhere and when I put it into the servername it says: `ERR: connect: connect->getConnection: client-socket error. Address lookup failed for host example.com`
when I leave it out it says: `ERR: connect: getSftpChannel: Unable to start subsystem: sftp`

any ideas how to solve this?

**Comment by marwie** [comment]
Hi, thanks for opening an issue. We'll look at exposing this option. I think exposing an extra optional port field would be more intuitive than server:port since it does'nt require any extra knowledge that this is possible to define.

**Reply by marwie** [reply]
@Sythelux the next update will expose a port option in the FTP server asset

**Reply by marwie** [reply]
@Sythelux this feature has been added in 3.37.6-pre just a few minutes ago https://discord.com/channels/717429793926283276/1011401449013526598/1233438678093791262

**Reply by marwie** [reply]
Please let me know once you had time to try it and if it solves your issue

**Reply by Sythelux** [reply]
yes, this works now. Thank you a lot!
